ts/features/wallet/onboarding/bancomatPay/screens/add-account/AddBPayScreen.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  getValueOrElse,
+  isError,
+  isLoading,
+  isReady
+} from "../../../../../bonus/bpd/model/RemoteValue";
+import {
+  addBPayToWallet,
+  walletAddBPayCancel,
+  walletAddBPayCompleted
+} from "../../store/actions";
+import {
+  onboardingBPayAddingResultSelector,
+  onboardingBPayChosenPanSelector
+} from "../../store/reducers/addingBPay";
+import { onboardingBPayFoundAccountsSelector } from "../../store/reducers/foundBpay";
+import LoadAddBPayComponent from "./LoadAddBPayComponent";
+
+type Props = ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps> &
+  Pick<React.ComponentProps<typeof BaseScreenComponent>, "contextualHelp">;
+
+type NextAction = {
+  index: number;
+  skip: boolean;
+};
+/**
+ * This screen is displayed when BPay are found and ready to be added in wallet
+ * @constructor
+ */
+const AddBPayScreen: React.FunctionComponent<Props> = (props: Props) => {
+  // next could be skip or not (a pan should be added)
+  const [currentAction, setNextAction] = React.useState<NextAction>({
+    index: 0,
+    skip: false
+  });
+
+  const currentIndex = currentAction.index;
+
+  React.useEffect(() => {
+    // call onCompleted when the end of pans has been reached
+    // and the adding phase has been completed (or it was skipped step)
+    if (
+      currentIndex >= props.bPayAccounts.length &&
+      (currentAction.skip || props.isAddingReady)
+    ) {
+      props.onCompleted();
+    }
+  }, [currentAction, props.isAddingReady]);
+
+  const nextPan = (skip: boolean) => {
+    const nextIndex = currentIndex + 1;
+    setNextAction({ index: nextIndex, skip });
+  };
+
+  const handleOnContinue = () => {
+    if (currentIndex < props.bPayAccounts.length) {
+      props.addBPay(props.bPayAccounts[currentIndex]);
+    }
+    nextPan(false);
+  };
+
+  const currentPan = index(currentIndex, [...props.bPayAccounts]);
+
+  return props.loading || props.isAddingResultError ? (
+    <LoadAddBPayComponent
+      isLoading={!props.isAddingResultError}
+      onCancel={props.onCancel}
+      onRetry={() => fromNullable(props.selectedBPay).map(props.onRetry)}
+    />
+  ) : currentPan.isSome() ? (
+    // TODO: Replace with Iterative add component
+    <SafeAreaView>
+      <Button onPress={handleOnContinue}>
+        <Label>ContinueTMP</Label>
+      </Button>
+    </SafeAreaView>
+  ) : null; // this should not happen
+};
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  addBPay: (bPay: BPay) => dispatch(addBPayToWallet.request(bPay)),
+  onCompleted: () => dispatch(walletAddBPayCompleted()),
+  onCancel: () => dispatch(walletAddBPayCancel()),
+  onRetry: (bPay: BPay) => dispatch(addBPayToWallet.request(bPay))
+});
+
+const mapStateToProps = (state: GlobalState) => {
+  const remoteBPay = onboardingBPayFoundAccountsSelector(state);
+  const addingResult = onboardingBPayAddingResultSelector(state);
+  const bPayAccounts = fromNullable(
```

Comment:
```suggestion
  const bPayAccounts = getValueOrElse(remoteBPay, []);
```
same but more simply

Reply:
changed in d360694c17874386ec20b8365e826b27b29a8e88